helix-core/src/main/java/org/apache/helix/PropertyKey.java

```diff
@@ -485,6 +486,52 @@ public PropertyKey currentState(String instanceName, String sessionId, String re
       }
     }
 
+    /**
+     * Get a property key associated with {@link CurrentState} of an instance and session. This key
```

Comment:
Both current state and task current state are using the same CurrentState. Logically it's fine, but it is really confusing. Is this the ultimate solution? Do we have plan to rename?

Reply:
Down the line, task CurrentStates will have a different model. It's not within the scope of this change, though.